include/pmacc/memory/boxes/SharedBox.hpp

```diff
@@ -111,12 +117,33 @@ namespace pmacc
         DINLINE static SharedBox init(T_Worker const& worker)
         {
             auto& mem_sh
-                = memory::shared::allocate<T_id, memory::Array<ValueType, math::CT::volume<Size>::type::value>>(
+                = memory::shared::allocate<T_id, memory::Array<ValueType, math::CT::volume<Extent>::type::value>>(
                     worker);
             return {mem_sh.data()};
         }
 
+    private:
+        /** get the pointer of the value relative to the origin pointer m_ptr
+         *
+         * @tparam T_MemoryIdxType Index type
+         * @param idx n-dimensional offset
+         * @return pointer to value
+         */
+        HDINLINE T_Type const* ptr(DataSpace<T_dim> const& idx) const
+        {
+            // offset in elements
+            int offset = idx.x();
+
+            if constexpr(Dim >= DIM2)
+            {
+                using Pitch = typename detail::CalcPitch<Extent, Extent::dim>::type;
+                for(uint32_t d = 1u; d < T_dim; ++d)
```

Comment:
This only iterates over `T_dim - 1` values. Is this intended?

Reply:
Yes is intended, pitch has one dimension less compared to the index.